es-app/src/guis/GuiGamelistOptions.cpp

```diff
@@ -89,12 +89,25 @@ GuiGamelistOptions::GuiGamelistOptions(Window* window, SystemData* system) : Gui
 			if(!hasFavorite)
 			{
 				LOG(LogDebug) << "Nothing to show in selected favorites mode, resetting";
-				favorite_only->setState(false);			
+				favorite_only->setState(false);
 				Settings::getInstance()->setBool("FavoritesOnly", favorite_only->getState());
 			}
 		}
 	});
-	
+
+	// Toggle: Show hidden
+	auto show_hidden = std::make_shared<SwitchComponent>(mWindow);
+	show_hidden->setState(Settings::getInstance()->getBool("ShowHidden"));
+	mMenu.addWithLabel("SHOW HIDDEN", show_hidden);
+	addSaveFunc([show_hidden, this] {
+		// First save to settings, in order for filtering to work
+		if (show_hidden->getState() != Settings::getInstance()->getBool("ShowHidden"))
+		{
+			Settings::getInstance()->setBool("ShowHidden", show_hidden->getState());
+			mHiddenStateChanged = true;
+		}
+	});
```

Comment:
after this, you'll need to check if there is anything left to show in the view, see above for the same situation in favorites_only.

I agree, that this is likely not an issue with hidden items, as it is with favorites or kid-only stuff, and really, this shouldnt even be handled in this file.
It should be checked with a specific function call to filedata if we are going to encounter this situation more often.
But untill then, it needs to be checked.